artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -590,9 +590,9 @@ private static ActiveMQMessage toAMQMessage(MessageReference reference,
       }
       amqMsg.setCommandId(commandId);
 
-      final SimpleString corrId = getObjectProperty(coreMessage, SimpleString.class, OpenWireConstants.JMS_CORRELATION_ID_PROPERTY);
-      if (corrId != null) {
-         amqMsg.setCorrelationId(corrId.toString());
+      final Object correlationID = coreMessage.getCorrelationID();
```

Comment:
It seems like we could retain support for carried byte[] type correlation IDs by decoding a UTF8 string here as the receiving Openwire client would either decode it if the getter for bytes was called or would just return a string value from the non-bytes getter, which allows basic interop.  Equally there are types that could be carried in an AMQP message correlation ID like a UUID that could make sense to convert to string as well as it could allow some edge case interop for receivers wanting to sync up correlation ID values.   It might make sense to have some consideration for those cases and what we should or shouldn't be allowing to propagate to an Openwire client.

Reply:
I updated this bit to support `byte[]` conversion to UTF8 `String`. At this point `UUID` is left out. We can add that later if necessary.

Reply:
That should be fine as the current AMQP to Core converter stringifies the correlation ID so it will carry through as a somewhat identifiable version of the original type.